static/js/place/overview.tsx

```diff
@@ -30,24 +32,66 @@ interface OverviewPropType {
   locale: string;
 }
 
-class Overview extends React.Component<OverviewPropType> {
+/**
+ * Holds response of /api/place/ranking, used to populate a ranking table.
+ */
+interface OverviewStateType {
+  rankingData: {
+    label: string[];
+    Population: { name: Record<string, unknown>; label: string }[];
+  };
+}
+class Overview extends React.Component<OverviewPropType, OverviewStateType> {
+  constructor(props: OverviewPropType) {
+    super(props);
+    this.state = {
+      rankingData: { label: [], Population: [] },
+    };
+  }
+
   render(): JSX.Element {
     return (
       <section className="factoid col-12">
         <div className="row">
-          <div className="col-12 col-md-4">
-            <GoogleMap dcid={this.props.dcid}></GoogleMap>
-          </div>
-          <div className="col-12 col-md-8">
-            <Ranking
-              dcid={this.props.dcid}
-              locale={this.props.locale}
-            ></Ranking>
+          <div
+            className={`col-12 ${this.shouldShowRanking() ? "col-md-4" : ""}`}
+          >
+            <div className={this.shouldShowRanking() ? "map-with-margin" : ""}>
+              <GoogleMap dcid={this.props.dcid}></GoogleMap>
+            </div>
           </div>
+          {this.shouldShowRanking() ? (
```

Comment:
can make this check once at the top of the render function & save it in a constant : `const shouldShowRanking = this.shouldShowRanking()`

Reply:
Ah, this is a much better idea than the helper function I tried to use instead! Switched to saving a constant at the top.